extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/impl/DbCheckingPMetaSupplier.java

```diff
@@ -0,0 +1,170 @@
+import static com.hazelcast.jet.mongodb.impl.MongoUtilities.checkDatabaseExists;
+
+/**
+ * A {@link ProcessorMetaSupplier} that will check if requested database and collection exist before creating
+ * the processors.
+ */
+public class DbCheckingPMetaSupplier implements ProcessorMetaSupplier {
+
+    private final Permission requiredPermission;
+    private final boolean shouldCheck;
+    private final ProcessorMetaSupplier standardForceOnePMS;
+    private boolean forceTotalParallelismOne;
+    private final String databaseName;
+    private final String collectionName;
+    private final ProcessorSupplier processorSupplier;
+    private final SupplierEx<? extends MongoClient> clientSupplier;
+    private final DataConnectionRef dataConnectionRef;
+    private int preferredLocalParallelism;
+
+    /**
+     * Creates a new instance of this meta supplier.
+     */
+    public DbCheckingPMetaSupplier(@Nullable Permission requiredPermission,
+                                   boolean shouldCheck,
+                                   boolean forceTotalParallelismOne,
+                                   @Nullable String databaseName,
+                                   @Nullable String collectionName,
+                                   @Nullable SupplierEx<? extends MongoClient> clientSupplier,
+                                   @Nullable DataConnectionRef dataConnectionRef,
+                                   @Nonnull ProcessorSupplier processorSupplier,
+                                   int preferredLocalParallelism
+    ) {
+        this.requiredPermission = requiredPermission;
+        this.shouldCheck = shouldCheck;
+        this.forceTotalParallelismOne = forceTotalParallelismOne;
+        this.databaseName = databaseName;
+        this.collectionName = collectionName;
+        this.processorSupplier = processorSupplier;
+        this.clientSupplier = clientSupplier;
+        this.dataConnectionRef = dataConnectionRef;
+        this.preferredLocalParallelism = forceTotalParallelismOne ? 1 : preferredLocalParallelism;
+        this.standardForceOnePMS = ProcessorMetaSupplier.forceTotalParallelismOne(processorSupplier);
+    }
+
+    @Override
+    public int preferredLocalParallelism() {
+        return forceTotalParallelismOne ? 1 : preferredLocalParallelism;
+    }
+
+    @Nullable
+    @Override
+    public Permission getRequiredPermission() {
+        return requiredPermission;
+    }
+
+    /**
+     * If true, only one instance of given supplier will be created.
+     */
+    public DbCheckingPMetaSupplier forceTotalParallelismOne(boolean forceTotalParallelismOne) {
+        this.forceTotalParallelismOne = forceTotalParallelismOne;
+        return this;
+    }
+
+    @Override
+    public boolean initIsCooperative() {
+        return !shouldCheck;
+    }
+
+    @Override
+    public void init(@Nonnull Context context) throws Exception {
+        if (forceTotalParallelismOne) {
+            preferredLocalParallelism = 1;
```

Comment:
`init` does not seem like a good place to change PMS config

Reply:
c19ee2aba108026791d6b63d7d043d5c7011328d